### ISSUE TITLE
For CV reports, `report.metrics.summarize(data_source="both")` returns misleading results.

### ISSUE THREAD

**Issue by cakedev0** [probabl-ai/skore#2546]
### Describe the bug

From the docstring, I read that `data_source="both"` should not be accepted for `CrossValidationReport.metrics.summarize`, but it actually is and returns wrong results (unweighted average of train & test, which doesn't make sense and is not something you want to look at).

I think it would make sense to accept it and to return something similar what's returned for `EstimatorReport` (concatenation of metrics of both sources):

| Metric           |   RandomForestRegressor (train) |   RandomForestRegressor (test) |
|:-----------------|--------------------------------:|-------------------------------:|
| R²               |                       0.803981  |                      0.807788  |
| RMSE             |                       0.219989  |                      0.22171   |
| Fit time (s)     |                       0.23386   |                      0.23386   |
| Predict time (s) |                       0.0230818 |                      0.0231872 |

### Steps/Code to Reproduce

```Python
import numpy as np

from sklearn.ensemble import RandomForestRegressor
from skore import CrossValidationReport, EstimatorReport

n = 10_000
X = np.random.rand(n, 5)
y = X[:, :3].sum(axis=1)

reg = RandomForestRegressor(random_state=0, min_impurity_decrease=0.01, n_estimators=50, n_jobs=2)
report = CrossValidationReport(reg, X, y)
report.metrics.summarize(data_source="both").frame()
```

### Expected Behavior

Concat the data of both sources

### Actual Behavior

Averages the data of both sources

### Environment

```shell
System:
    python: 3.12.11 (main, Aug 18 2025, 19:19:11) [Clang 20.1.4 ]
executable: /home/arthur/probabl/skore/skore/.mlflow-venv/bin/python
   machine: Linux-6.17.0-14-generic-x86_64-with-glibc2.39

Python dependencies:
        skore: 0.0.0+unknown
          pip: 26.0.1
       jinja2: 3.1.6
       joblib: 1.5.3
   matplotlib: 3.10.8
        numpy: 2.4.2
       pandas: 2.3.3
       plotly: 6.5.1
         rich: 14.3.3
 scikit-learn: 1.8.0
      seaborn: 0.13.2
 skore[local]: None
        skrub: 0.7.1
skore-hub-project: None
```

**Comment by glemaitre** [comment]
Indeed. Maybe we should quickly fix it with a `NotImplementedError` and come back to implement the feature once the refactor related to https://github.com/probabl-ai/skore/pull/2536 is implemented.
